library/core/src/ptr/const_ptr.rs

```diff
@@ -394,6 +394,37 @@ impl<T: ?Sized> *const T {
     where
         T: Sized,
     {
+        #[inline]
+        const fn runtime_offset_nowrap(this: *const (), count: isize, size: usize) -> bool {
+            #[inline]
+            fn runtime(this: *const (), count: isize, size: usize) -> bool {
+                let Some(byte_offset) = count.checked_mul(size as isize) else {
+                    return false;
+                };
+                if byte_offset < 0 {
+                    -byte_offset as usize <= this.addr()
+                } else {
+                    this.addr().checked_add(byte_offset as usize).is_some()
+                }
+            }
+
+            const fn comptime(_: *const (), _: isize, _: usize) -> bool {
+                true
+            }
+
+            intrinsics::const_eval_select((this, count, size), comptime, runtime)
```

Comment:
This needs the usual comment about why it is okay to have different behavior for compile time and runtime.